sat/cli/bootprep/image.py

```diff
@@ -451,18 +444,53 @@ def create_images(instance, args):
     # Do this early to provide information required to render names.
     validate_image_ims_bases(input_images)
 
-    # Raises ImageCreateError if validation of CFS configurations fails
-    validate_image_configurations(input_images, cfs_client,
-                                  instance.input_configuration_names, args.dry_run)
+    # This validation is not necessary when images are skipped
+    if IMAGES_KEY in args.limit:
+        # When the configurations are not excluded via --limit, and it's a dry run,
+        # assume the configurations from the input file do exist.
+        input_configs_exist = CONFIGURATIONS_KEY in args.limit and args.dry_run
+        # Raises ImageCreateError if validation of CFS configurations fails
+        validate_image_configurations(input_images, cfs_client,
+                                      instance.input_configuration_names, input_configs_exist)
 
     # This is where name properties are first accessed and thus rendered
     validate_unique_image_names(input_images)
 
+    # TODO: Note the slight change in behavior here. Test this use case.
```

Comment:
If I'm understanding this correctly, I'm not too worried about the change in behavior here. We're checking to make sure that an image in the config isn't accidentally the base of another image, and previously we would have allowed the case in which the conflicting image was skipped? I can't think of a case in which this couldn't be resolved just by setting up the ref_name / image_ref properties correctly. Even if there are cases that worked previously that will start failing validation now, they probably aren't what the user intended anyway.

Reply:
That's right. Yeah, that's what I was thinking. I would still like to try to test this use case to make sure it doesn't blow up. I meant to do that, but I forgot. I will then probably remove this TODO comment.

Reply:
Here are the results of a test of this use case, showing before and after behavior for this case: https://gist.github.com/haasken-hpe/7cde42a46cb7a8c8766f88db902f940e

I think you could make a case for either behavior being correct, and it's a pretty specific edge case, so I think it's fine the way it is now with this change. I'll remove the comment.